FinSearchUnified/Helper/StaticHelper.php

```diff
@@ -696,8 +699,7 @@ public static function removeControlCharacters($value)
     public static function useShopSearch()
     {
         return (
-            !self::isFindologicActive() ||
-            self::checkDirectIntegration() ||
+            !self::isFindologicActive() || self::checkDirectIntegration() || Shopware()->Front()->Request() == null ||
```

Comment:
* Check for the request object right at the beginning. We actually don't care about the rest if there is no request anyway.
* Use `===`. Always.

Reply:
moving this in the beginning would cause error in the latter conditions as we are using `expects` in the mock object to use `atleastOnce`. We would have to remove that check or use `any` otherwise so it does not check the number of times the method is called

Reply:
@zep-com 
Unless you extract the case into a separate test?

Reply:
@zep-com 
Yes, definitely but our experience shows that it is better to split such things up as it makes the tests more understandable and easier to maintain.
While your argument is definitely valid, you will have to trust me that it is better to have separate tests in the long run.
Also, since the method returns early for this case, you won't have to create as much mocks.